src/schema/v2/artwork/index.ts

```diff
@@ -1891,33 +1892,56 @@ const Artwork: GraphQLFieldConfig<void, ResolverContext> = {
   resolve: async (
     _source,
     args,
-    { artworkLoader, marketPriceInsightsBatchLoader },
+    { artworkLoader, marketPriceInsightsBatchLoader, principalDirectivePath },
     resolveInfo
   ) => {
-    const { id } = args
-
-    const hasRequestedPriceInsights = isFieldRequested(
-      "marketPriceInsights",
-      resolveInfo
-    )
-
-    const artwork = await artworkLoader(id)
-
-    // // We don't want to query for the price insights unless the user has requested them
-    if (
-      marketPriceInsightsBatchLoader &&
-      artwork &&
-      hasRequestedPriceInsights
-    ) {
-      const enrichedArtworks = await enrichArtworksWithPriceInsights(
-        [artwork],
-        marketPriceInsightsBatchLoader
+    try {
+      const { id } = args
+
+      const hasRequestedPriceInsights = isFieldRequested(
+        "marketPriceInsights",
+        resolveInfo
       )
 
-      return enrichedArtworks?.[0]
-    }
+      const artwork = await artworkLoader(id)
+
+      // // We don't want to query for the price insights unless the user has requested them
+      if (
+        marketPriceInsightsBatchLoader &&
+        artwork &&
+        hasRequestedPriceInsights
+      ) {
+        const enrichedArtworks = await enrichArtworksWithPriceInsights(
+          [artwork],
+          marketPriceInsightsBatchLoader
+        )
 
-    return artwork
+        return enrichedArtworks?.[0]
+      }
+
+      return artwork
+    } catch (error) {
+      // When the upstream service returns a 404, and the `principalField`
+      // directive was not applied to this selection, return a partial
+      // object to allow any underlying field selections to still be resolved.
+      const isError404 = error.statusCode === 404
+      const responsePathsArray = responsePathAsArray(resolveInfo.path)
+      const isPrincipalField = _.isEqual(
+        responsePathsArray,
+        principalDirectivePath
+      )
+
+      if (isError404 && !isPrincipalField) {
+        return {
```

Comment:
There were a bunch of places in this file and included helpers (overall many lines of code) that have some implicit assumptions thrown around w.r.t to some data being present in artwork JSON- such as `artists` being exist-y in an artwork response, `images`, and so forth.

Rather than try to painstakingly add things like `if (!images?.length) return []` throughout, instead let's have a reasonable 'default' schema.

Locally, these defaults were able to get Force's fairly extensive artwork page query to return `"artwork": ...` without any top-level errors (which cause `"artwork": null`) for an unpublished artwork.

Reply:
This is also where we can merge in some of the minimal data Gravity might start to provide to the defaults, eventually from https://github.com/artsy/gravity/pull/17283

Reply:
So this change allows us to safely drop the `@prinicipalField` directive [from here](https://github.com/artsy/force/blob/706dafbb1f9e64557ec5c3d23c5b5a3196679931/src/Apps/Artwork/artworkRoutes.tsx#L26) (but not necessarily other usages of the `artwork` field), yeah?